### PR TITLE
Grid keyboard handling refactoring and bugfixes

### PR DESCRIPTION
**Pull request by ararus**
Keyboard navigation refactoring
Space and arrows don't scroll the page
Cell editors fixed (when edit mode initiated by typing)
Space on checkbox/radio column doesn't reset previous selection (it toggles selection on the focused row as expected)
API updated: onKeyDown prop added to GridColumn

### CODE REVIEW

File: packages/grid/src/Grid.css [hunk 1]
```diff
@@ -143,6 +143,10 @@
   --grid-rowSeparator-color: var(--uitkGrid-rowSeparator-color, var(--grid-rowSeparator-color-default));
   --grid-columnSeparator-color: var(--uitkGrid-columnSeparator-color, var(--grid-columnSeparator-color-default));
   --grid-rowSeparator-color-divided: var(--uitkGrid-rowSeparator-color-divided, var(--grid-rowSeparator-color-divided-default));
+
+  --grid-columnGhost-borderColor: var(--uitkGrid-columnGhost-borderColor, var(--uitk-palette-interact-border-hover));
```
Comment (libertymayc) [inline packages/grid/src/Grid.css:147]: does --uitk-selectable-borderColor-hover work here?

File: packages/grid/src/RowSelectionCheckboxColumn.tsx [hunk 2]
```diff
@@ -10,13 +12,27 @@ export type RowSelectionCheckboxColumnProps<T> = Omit<
...
   props: RowSelectionCheckboxColumnProps<T>
 ) {
+  const { selectRows } = useSelectionContext();
+
+  const onKeyDown = (
+    event: KeyboardEvent<HTMLDivElement>,
+    rowIndex: number
+  ) => {
+    if (event.key === " ") {
+      selectRows(rowIndex, false, true);
```
Comment (ZarkoPernar) [inline packages/grid/src/RowSelectionCheckboxColumn.tsx:22]: Can you refactor this to a `selectRows({rowIndex, whateverThisIs: false, andThat: true});`? As soon as I have more than one param, I usually refactor to object param, especially when I have boolean flags.

**Review by ZarkoPernar — APPROVED** [review]
You can make the refactor I noted in the next PR if you prefer 👍

File: packages/grid/src/Grid.css (hunk 1 above)
Comment (ararus) [reply packages/grid/src/Grid.css:147]: Darrin is working on characteristics for grids. Not sure if selectable is the best option here.

File: packages/grid/src/RowSelectionCheckboxColumn.tsx (hunk 2 above)
Comment (ararus) [reply packages/grid/src/RowSelectionCheckboxColumn.tsx]: Done

**Review by ZarkoPernar — APPROVED** [review]

**Pull request merged**